aspnetcore/security/authentication/customize-identity-model.md

```diff
@@ -0,0 +1,976 @@
+public abstract class IdentityUserContext<
+    TUser, TKey, TUserClaim, TUserLogin, TUserToken> : DbContext
+        where TUser : IdentityUser<TKey>
+        where TKey : IEquatable<TKey>
+        where TUserClaim : IdentityUserClaim<TKey>
+        where TUserLogin : IdentityUserLogin<TKey>
+        where TUserToken : IdentityUserToken<TKey>
+{
+}
+```
+
+## Customize the model
+
+The starting point for model customization is to derive from the appropriate context type. See the [Model generic types](#model-generic-types) section. This context type is customarily called `ApplicationDbContext` and is created by the ASP.NET Core templates.
+
+The context is used to configure the model in two ways:
+
+* Supplying entity and key types for the generic type parameters.
+* Overriding `OnModelCreating` to modify the mapping of these types.
+
+When overriding `OnModelCreating`, `base.OnModelCreating` should be called first; the overriding configuration should be called next. EF Core generally has a last-one-wins policy for configuration. For example, if the `ToTable` method for an entity type is called first with one table name and then again later with a different table name, the table name in the second call is used.
+
+### Use a custom User type
+
+To use a custom User type, create the type and have it inherit from `IdentityUser`. It's customary to name this type `ApplicationUser`. This type typically has additional properties not included in the base type, otherwise there would be no value in creating it. For example:
+
+```csharp
+public class ApplicationUser : IdentityUser
+{
+    public string CustomTag { get; set; }
+}
+```
+
+Next, use the `ApplicationUser` type as a generic argument for the context:
+
+```csharp
+public class ApplicationDbContext : IdentityDbContext<ApplicationUser>
+{
+    public ApplicationDbContext(DbContextOptions<ApplicationDbContext> options)
+        : base(options)
+    {
+    }
+}
+```
+
+There's no need to override `OnModelCreating` in the `ApplicationDbContext` class. EF Core maps the `CustomTag` property by convention. However, the database needs to be updated to create a new `CustomTag` column. To create the column, add a migration, and then update the database as described in [Identity and EF Core Migrations](#identity-and-ef-core-migrations).
+
+Update `Startup.ConfigureServices` to use the new `ApplicationUser` class:
+
+::: moniker range=">= aspnetcore-2.1"
+
+```csharp
+services.AddDefaultIdentity<ApplicationUser>()
+        .AddEntityFrameworkStores<ApplicationDbContext>()
+        .AddDefaultUI();
+```
+
+In ASP.NET Core 2.1 or later, Identity is provided as a Razor Class Library. For more information, see <xref:security/authentication/scaffold-identity>. Consequently, the preceding code requires a call to <xref:Microsoft.AspNetCore.Identity.IdentityBuilderUIExtensions.AddDefaultUI*>. If the Identity scaffolder was used to add Identity files to the project, remove the call to `AddDefaultUI`.
```

Comment:
@HaoK Thoughts on this?

Reply:
It probably doesn't hurt to call out, its a bit of a mess since sometimes its not always clear when AddDefaultUI is needed or not, it depends on how much they scaffold I think... And I believe its always safe to have

Reply:
Sounds good.